packages/video_player_avplayer/tizen/src/video_player_tizen_plugin.cc

```diff
@@ -0,0 +1,340 @@
+  int32_t drm_type = 0;  // DRM_TYPE_NONE
+  std::string license_server_url;
+  bool prebuffer_mode;
+  std::string format;
+
+  if (msg.asset() && !msg.asset()->empty()) {
+    char *res_path = app_get_resource_path();
+    if (res_path) {
+      uri = uri + res_path + "flutter_assets/" + *msg.asset();
+      free(res_path);
+    } else {
+      return FlutterError("Internal error", "Failed to get resource path.");
+    }
+  } else if (msg.uri() && !msg.uri()->empty()) {
+    uri = *msg.uri();
+    if (msg.format_hint() && !msg.format_hint()->empty()) {
+      format = *msg.format_hint();
+    }
+
+    const flutter::EncodableMap *drm_configs = msg.drm_configs();
+    if (drm_configs) {
+      auto iter = drm_configs->find(flutter::EncodableValue("drmType"));
+      if (iter != drm_configs->end()) {
+        if (std::holds_alternative<int32_t>(iter->second)) {
+          drm_type = std::get<int32_t>(iter->second);
+        }
+      }
+      iter = drm_configs->find(flutter::EncodableValue("licenseServerUrl"));
+      if (iter != drm_configs->end()) {
+        if (std::holds_alternative<std::string>(iter->second)) {
+          license_server_url = std::get<std::string>(iter->second);
+        }
+      }
+    }
+
+    const flutter::EncodableMap *player_options = msg.player_options();
+    if (player_options) {
+      auto iter =
+          player_options->find(flutter::EncodableValue("prebufferMode"));
+      if (iter != player_options->end()) {
+        if (std::holds_alternative<bool>(iter->second)) {
+          prebuffer_mode = std::get<bool>(iter->second);
+        }
+      }
+    }
+  } else {
+    return FlutterError("Invalid argument", "Either asset or uri must be set.");
+  }
+
+  int64_t player_id = 0;
+  if (uri.substr(0, 4) == "http") {
+    auto player = std::make_unique<PlusPlayer>(plugin_registrar_->messenger(),
+                                               native_window, format);
+    player_id =
+        player->Create(uri, drm_type, license_server_url, prebuffer_mode);
+    players_[player_id] = std::move(player);
+  } else {
+    auto player = std::make_unique<MediaPlayer>(plugin_registrar_->messenger(),
+                                                native_window);
+    player_id =
+        player->Create(uri, drm_type, license_server_url, prebuffer_mode);
+    players_[player_id] = std::move(player);
+  }
+
+  PlayerMessage result(player_id);
+  return result;
+}
+
+std::optional<FlutterError> VideoPlayerTizenPlugin::Dispose(
+    const PlayerMessage &msg) {
+  auto iter = players_.find(msg.player_id());
+  if (iter != players_.end()) {
+    iter->second->Dispose();
+    players_.erase(iter);
+  }
+  return std::nullopt;
+}
+
+std::optional<FlutterError> VideoPlayerTizenPlugin::SetLooping(
+    const LoopingMessage &msg) {
+  VideoPlayer *player = FindPlayerById(msg.player_id());
+  if (!player) {
+    return FlutterError("Invalid argument", "Player not found.");
+  }
+  player->SetLooping(msg.is_looping());
```

Comment:
The operation of player may be failed, it should not always return 'std::nullopt'.
Please throw an error in player when operation failed, use 'try' and 'catch' to get the error.
I think all of player operations should process error.

Reply:
Fixed.